src/main/java/de/tum/in/www1/artemis/web/rest/FileUploadSubmissionResource.java

```diff
@@ -33,81 +46,327 @@
      */
-    @GetMapping("/file-upload-submissions")
-    public List<FileUploadSubmission> getAllFileUploadSubmissions() {
-        log.debug("REST request to get all FileUploadSubmissions");
-        return fileUploadSubmissionRepository.findAll();
+    @GetMapping("/file-upload-submissions/{submissionId}")
+    public ResponseEntity<FileUploadSubmission> getFileUploadSubmission(@PathVariable Long submissionId) {
+        log.debug("REST request to get FileUploadSubmission with id: {}", submissionId);
+        var fileUploadExercise = (FileUploadExercise) fileUploadSubmissionService.findOne(submissionId).getParticipation().getExercise();
+        if (!authCheckService.isAtLeastTeachingAssistantForExercise(fileUploadExercise)) {
+            return forbidden();
+        }
+        var fileUploadSubmission = fileUploadSubmissionService.getLockedFileUploadSubmission(submissionId, fileUploadExercise);
+        // Make sure the exercise is connected to the participation in the json response
+        StudentParticipation studentParticipation = (StudentParticipation) fileUploadSubmission.getParticipation();
+        studentParticipation.setExercise(fileUploadExercise);
+        hideDetails(fileUploadSubmission);
+        return ResponseEntity.ok(fileUploadSubmission);
     }
 
     /**
-     * GET /file-upload-submissions/:id : get the "id" fileUploadSubmission.
+     * GET /file-upload-submissions : get all the fileUploadSubmissions for an exercise. It is possible to filter, to receive only the one that have been already submitted, or only the one
+     * assessed by the tutor who is doing the call
      *
-     * @param id the id of the fileUploadSubmission to retrieve
-     * @return the ResponseEntity with status 200 (OK) and with body the fileUploadSubmission, or with status 404 (Not Found)
+     * @param exerciseId the id of the exercise
+     * @param submittedOnly if only submitted submissions should be returned
+     * @param assessedByTutor if the submission was assessed by calling tutor
+     * @return the ResponseEntity with status 200 (OK) and the list of File Upload Submissions in body
      */
-    @GetMapping("/file-upload-submissions/{id}")
-    public ResponseEntity<FileUploadSubmission> getFileUploadSubmission(@PathVariable Long id) {
-        log.debug("REST request to get FileUploadSubmission : {}", id);
-        Optional<FileUploadSubmission> fileUploadSubmission = fileUploadSubmissionRepository.findById(id);
-        return ResponseUtil.wrapOrNotFound(fileUploadSubmission);
+    @GetMapping("/exercises/{exerciseId}/file-upload-submissions")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<List<FileUploadSubmission>> getAllFileUploadSubmissions(@PathVariable Long exerciseId, @RequestParam(defaultValue = "false") boolean submittedOnly,
+            @RequestParam(defaultValue = "false") boolean assessedByTutor) {
```

Comment:
Also here there are a lot of similiarities to e.g. the `TextSubmissionResource`. Why do we duplicate these methods for every new exercise type?

Reply:
As discussed we refactor this in a later PR